src/java.base/share/classes/jdk/internal/classfile/attribute/NestMembersAttribute.java

```diff
@@ -0,0 +1,84 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.attribute;
+
+import java.lang.constant.ClassDesc;
+import java.util.Arrays;
+import java.util.List;
+
+import jdk.internal.classfile.Attribute;
+import jdk.internal.classfile.ClassElement;
+import jdk.internal.classfile.constantpool.ClassEntry;
+import jdk.internal.classfile.impl.BoundAttribute;
+import jdk.internal.classfile.impl.UnboundAttribute;
+import jdk.internal.classfile.impl.Util;
+
+/**
+ * Models the {@code NestMembers} attribute (JVMS 4.7.29), which can
+ * appear on classes to indicate that this class is the host of a nest.
+ * Delivered as a {@link jdk.internal.classfile.ClassElement} when
+ * traversing the elements of a {@link jdk.internal.classfile.ClassModel}.
+ */
+public sealed interface NestMembersAttribute extends Attribute<NestMembersAttribute>, ClassElement
+        permits BoundAttribute.BoundNestMembersAttribute, UnboundAttribute.UnboundNestMembersAttribute {
+
+    /**
+     * {@return the classes belonging to the nest hosted by this class}
+     */
+    List<ClassEntry> nestMembers();
+
+    /**
+     * {@return a {@code NestMembers} attribute}
+     * @param nestMembers the member classes of the nest
+     */
+    static NestMembersAttribute of(List<ClassEntry> nestMembers) {
+        return new UnboundAttribute.UnboundNestMembersAttribute(nestMembers);
+    }
+
+    /**
+     * {@return a {@code NestMembers} attribute}
+     * @param nestMembers the member classes of the nest
+     */
+    static NestMembersAttribute of(ClassEntry... nestMembers) {
+        return of(List.of(nestMembers));
+    }
+
+    /**
+     * {@return a {@code NestMembers} attribute}
+     * @param nestMembers the member classes of the nest
+     */
+    static NestMembersAttribute ofSymbols(List<ClassDesc> nestMembers) {
```

Comment:
I see that in some classes we use the `Symbols` prefix, in others we just use `of` - we should try to make the more consistent (in the future).

Reply:
`ofSymbols` is an alternative to `of` when conflicting method parameters.
In such case `of` refers to CP entries and `ofSymbols` refer to independent symbols describing the objects, like for example `ClassDesc`, `MethodTypeDesc`, `PackageDesc`

Reply:
> `ofSymbols` is an alternative to `of` when conflicting method parameters. In such case `of` refers to CP entries and `ofSymbols` refer to independent symbols describing the objects, like for example `ClassDesc`, `MethodTypeDesc`, `PackageDesc`

Yes, I was noting that some classes e.g. ModuleProvideInfo do not seem to follow this logic:

```
static ModuleProvideInfo of(ClassEntry provides, ClassEntry... providesWith) { ... }

static ModuleProvideInfo of(ClassDesc provides, List<ClassDesc> providesWith) { ... }
```

Reply:
Yes, the alternate `ofSymbols` is used only when there is a conflict, which is not the case of `ModuleProvideInfo` factory methods, there is differentiating first argument.
In case of var-args vs Lists - that is another topic with long history of search for the right balance.